frontend/src/components/widgets/FileUploader/FileUploader.tsx

```diff
@@ -77,7 +81,36 @@ class FileUploader extends React.PureComponent<Props, State> {
 
   public constructor(props: Props) {
     super(props)
-    this.state = { files: [], newestServerFileId: 0 }
+    this.state = this.initialValue
+  }
+
+  get initialValue(): State {
+    const emptyState = { files: [], newestServerFileId: 0 }
+    const { widgetMgr, element } = this.props
+
+    const widgetValue = widgetMgr.getFileUploaderStateValue(element)
+    if (!widgetValue) {
+      return emptyState
+    }
+
+    const { maxFileId, uploadedFileInfo } = widgetValue
+    if (!maxFileId || !uploadedFileInfo) {
```

Comment:
Nit: check against `null`/`undefined` (or `0`, if that's appropriate), instead of using truthy/falsy checks

Reply:
👍 done